plothist/comparison.py

```diff
@@ -201,38 +200,96 @@ def get_difference(h1, h2, h1_uncertainty_type="symmetrical"):
 
     Returns
     -------
-    comparison_values : numpy.ndarray
+    difference_values : numpy.ndarray
         The difference values.
-    comparison_uncertainties_low : numpy.ndarray
+    difference_uncertainties_low : numpy.ndarray
         The lower uncertainties on the difference.
-    comparison_uncertainties_high : numpy.ndarray
+    difference_uncertainties_high : numpy.ndarray
         The upper uncertainties on the difference.
     """
     _check_uncertainty_type(h1_uncertainty_type)
     _check_binning_consistency([h1, h2])
 
-    comparison_values = h1.values() - h2.values()
+    difference_values = h1.values() - h2.values()
 
     if h1_uncertainty_type == "asymmetrical":
         uncertainties_low, uncertainties_high = get_asymmetrical_uncertainties(h1)
 
-        comparison_uncertainties_low = np.sqrt(
-            uncertainties_low ** 2 + h2.variances()
-        )
-        comparison_uncertainties_high = np.sqrt(
-            uncertainties_high ** 2 + h2.variances()
+        difference_uncertainties_low = np.sqrt(uncertainties_low**2 + h2.variances())
+        difference_uncertainties_high = np.sqrt(
+            uncertainties_high**2 + h2.variances()
         )
     else:
-        comparison_uncertainties_low = np.sqrt(h1.variances() + h2.variances())
-        comparison_uncertainties_high = comparison_uncertainties_low
+        difference_uncertainties_low = np.sqrt(h1.variances() + h2.variances())
+        difference_uncertainties_high = difference_uncertainties_low
 
     return (
-        comparison_values,
-        comparison_uncertainties_low,
-        comparison_uncertainties_high,
+        difference_values,
+        difference_uncertainties_low,
+        difference_uncertainties_high,
     )
 
 
+def get_efficency(h1, h2):
+    """
+    Calculate the ratio of two correlated histograms (h1/h2), in which the entries of h1 are a subsample of the entries of h2.
+    The variances are calculated according to the formula given in :ref:`documentation-statistics-label`.
+
+    The following conditions must be fulfilled for the calculation of the efficiency:
+    * The bins of the histograms must be equal.
+    * The histograms must be unweighted.
```

Comment:
What about the weighted case?

I think it has been implemented here: https://github.com/scikit-hep/hist/blob/ec2842326046bf860b12646222aa1d4ed45c2092/src/hist/intervals.py#L80

Reply:
I am not sure this covers the weighted case, this is just another approach. Since our approach is already less naïve than the most naïve one that is usually used: eff(1-eff)/N, so I would say this is a good compromise for now.